code/modules/mob/living/carbon/xenomorph/castes/king/abilities_king.dm

```diff
@@ -32,104 +32,359 @@
 
 
 // ***************************************
-// *********** Gravity Crush
+// *********** Petrify
 // ***************************************
-#define WINDUP_GRAV 1 SECONDS
-
-/datum/action/xeno_action/activable/gravity_crush
-	name = "Gravity Crush"
-	action_icon_state = "fortify"
-	mechanics_text = "Increases the localized gravity in an area and crushes everything in it."
-	ability_name = "Gravity crush"
-	plasma_cost = 200
+#define PETRIFY_RANGE 7
+#define PETRIFY_DURATION 6 SECONDS
+#define PETRIFY_WINDUP_TIME 2 SECONDS
+/datum/action/xeno_action/petrify
+	name = "Petrify"
+	action_icon_state = "petrify"
+	mechanics_text = "After a windup, petrifies all humans looking at you. While petrified humans are immune to damage, but also can't attack."
+	ability_name = "petrify"
+	plasma_cost = 100
 	cooldown_timer = 30 SECONDS
+	keybind_flags = XACT_KEYBIND_USE_ABILITY
 	keybinding_signals = list(
-		KEYBINDING_NORMAL = COMSIG_XENOABILITY_GRAVITY_CRUSH,
+		KEYBINDING_NORMAL = COMSIG_XENOABILITY_PETRIFY,
 	)
-	/// How far can we use gravity crush
-	var/king_crush_dist = 5
-	/// A list of all things that had a fliter applied
-	var/list/filters_applied = list()
 
-/datum/action/xeno_action/activable/gravity_crush/on_cooldown_finish()
-	to_chat(owner, span_warning("Our psychic aura restores itself. We are ready to gravity crush again."))
-	return ..()
+/datum/action/xeno_action/petrify/action_activate()
+	var/obj/effect/overlay/eye/eye = new
+	owner.vis_contents += eye
+	flick("eye_opening", eye)
+	playsound(owner, 'sound/effects/petrify_charge.ogg', 50)
+	if(!do_after(owner, PETRIFY_WINDUP_TIME, FALSE, owner, BUSY_ICON_DANGER))
+		flick("eye_closing", eye)
+		addtimer(CALLBACK(src, .proc/remove_eye, eye), 7, TIMER_CLIENT_TIME)
+		return
+	playsound(owner, 'sound/effects/petrify_activate.ogg', 50)
+	var/list/mob/living/carbon/human/humans = list()
+	for(var/mob/living/carbon/human/human in view(PETRIFY_RANGE, owner.loc))
+		if(is_blind(human))
+			continue
+		humans += human
+		human.notransform = TRUE
+		human.status_flags |= GODMODE
+		human.add_atom_colour(COLOR_GRAY, TEMPORARY_COLOUR_PRIORITY)
+		ADD_TRAIT(human, TRAIT_HANDS_BLOCKED, REF(src))
+		human.move_resist = MOVE_FORCE_OVERPOWERING
+		human.log_message("has been petrified by [owner] for [PETRIFY_DURATION] ticks", LOG_ATTACK, color="pink")
+		var/image/stone_overlay = image('icons/effects/effects.dmi', null, "petrified_overlay")
+		stone_overlay.filters += filter(arglist(alpha_mask_filter(render_source="*[REF(human)]",flags=MASK_INVERSE)))
+		var/mutable_appearance/mask = mutable_appearance()
+		mask.appearance = human.appearance
+		mask.render_target = "*[REF(human)]"
+		mask.alpha = 125
+		stone_overlay.overlays += mask
+		human.overlays += stone_overlay
+		humans[human] = stone_overlay
```

Comment:
Try to add some lines between logic blocks, so it's not a wall of text

Reply:
Tidied it up.